By default GNOME folks want us to put all stuff in /usr/share/gdm/env.d
(which is a lot of effort for no good reason for us) or use the systemd
specific /etc/environment.d to set environment variables. With this
patch we just force gnome-session's hand to include the env variables
of /etc/profile.

--- a/gnome-session/gnome-session.in
+++ b/gnome-session/gnome-session.in
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
 if [ "x$XDG_SESSION_TYPE" = "xwayland" ] &&
    [ "x$XDG_SESSION_CLASS" != "xgreeter" ] &&
@@ -25,4 +25,6 @@ if [ -n "$REGION" ]; then
   export LC_PAPER=$REGION
 fi
 
+source /etc/profile
+
 exec @libexecdir@/gnome-session-binary "$@"
